src/expandableCalendar/agendaList.tsx

```diff
@@ -1,27 +1,54 @@
 import _ from 'lodash';
-import React, {Component} from 'react';
-import {SectionList, Text} from 'react-native';
 import PropTypes from 'prop-types';
 import XDate from 'xdate';
-import {isToday as dateutils_isToday} from '../dateutils';
+
+import React, {Component} from 'react';
+import {Text, SectionList, SectionListProps, DefaultSectionT, SectionListData, ViewStyle, NativeSyntheticEvent, NativeScrollEvent, LayoutChangeEvent, ViewToken} from 'react-native';
+
+// @ts-expect-error
+import {isToday} from '../dateutils';
+// @ts-expect-error
+import {getMoment} from '../momentResolver';
+import {Theme} from '../types';
 import styleConstructor from './style';
 import asCalendarConsumer from './asCalendarConsumer';
-import {getMoment} from '../momentResolver';
+
 
 const commons = require('./commons');
-const UPDATE_SOURCES = commons.UPDATE_SOURCES;
+const updateSources = commons.UPDATE_SOURCES;
+
+interface Props extends SectionListProps<any, DefaultSectionT> {
+  /** day format in section title. Formatting values: http://arshaw.com/xdate/#Formatting */
+  dayFormat?: string;
+  /** a function to custom format the section header's title */
+  dayFormatter?: (arg0: string) => string;
+  /** whether to use moment.js for date string formatting
+   * (remember to pass 'dayFormat' with appropriate format, like 'dddd, MMM D') */
+  useMoment?: boolean;
+  /** whether to mark today's title with the "Today, ..." string. Default = true */
+  markToday?: boolean;
+  /** style passed to the section view */
+  sectionStyle?: ViewStyle;
+  /** whether to block the date change in calendar (and calendar context provider) when agenda scrolls */
+  avoidDateUpdates?: boolean;
+  /** offset scroll to section */
+  viewOffset?: number;
+  theme?: Theme;
+  context?: any;
+}
+export type AgendaListProps = Props;
```

Comment:
I like it clean. Meaning, pass `Props` in the class instead of this long name so it is clearer for developers that look at the code, and only export the named one for the users to use.